src/merkle/tiered_smt/mod.rs

```diff
@@ -274,6 +274,10 @@ impl TieredSmt {
         })
     }
 
+    pub fn upper_leaves_by_tier(&self) -> impl Iterator<Item = (&NodeIndex, &RpoDigest)> {
+        self.nodes.upper_leaves()
+    }
```

Comment:
nit: I'd name this `upper_leaf_nodes()`.

Also, since this is a part of the public interface, I'd add doc comments here.